hs_core/tests/api/native/test_publish_resource.py

```diff
@@ -35,20 +32,40 @@ def setUp(self):
         )
 
     def tearDown(self):
+        super(TestPublishResource, self).tearDown()
         self.user.uaccess.delete()
         User.objects.all().delete()
         Group.objects.all().delete()
         GenericResource.objects.all().delete()
 
     def test_publish_resource(self):
-        # publish resource
+        # check status prior to publishing the resource
+        self.assertFalse(
+            self.res.raccess.published,
+            msg='The resource is published'
+        )
+
+        self.assertFalse(
+            self.res.raccess.immutable,
+            msg='The resource is frozen'
+        )
+
+        self.assertIsNone(
+            self.res.doi,
+            msg='doi is assigned'
+        )
+
+        # there should not be published date type metadata element
+        self.assertFalse(self.res.metadata.dates.filter(type='published').exists())
+
+        # publish resource - this is the api we are testing
         hydroshare.publish_resource(self.res.short_id)
```

Comment:
@pkdash Sounds good, but could you add @unittest.skip for this test_publish_resource for now so that we don't accidentally run this test when my resource publication branch is merged in? We can then work on mocking up crossref and enable this test subsequently afterwards. How does this sound?